lintcheck/src/json.rs

```diff
@@ -130,9 +143,65 @@ pub(crate) fn diff(old_path: &Path, new_path: &Path) {
     println!();
     println!();
 
-    print_warnings("Added", &added);
-    print_warnings("Removed", &removed);
-    print_changed_diff(&changed);
+    print_warnings("Added", added);
+    print_warnings("Removed", removed);
+    print_changed_diff(changed);
+}
+
+/// This function limits the number of lints in the collection if needed.
+///
+/// It's honestly amazing that a function like this is needed. But some restriction
+/// lints (Looking at you `implicit_return` and `if_then_some_else_none`) trigger a lot
+/// like 60'000+ times in our CI.
+///
+/// Each lint in the list will be limited to 200 messages
+fn limit_by_lint<T, F>(list: &mut Vec<T>, lint_key: F) -> Option<String>
+where
+    F: Fn(&T) -> &str,
+{
+    const PER_LINT_LIMIT: usize = 200;
+
+    // Everything is fine, nothing to see here.
+    if list.len() < PER_LINT_LIMIT {
+        return None;
+    }
+
+    let mut keyed_list: Vec<_> = list
+        .drain(..)
+        .map(|warning| (lint_key(&warning).to_string(), warning))
+        .collect();
+
+    keyed_list.sort_unstable_by(|(a, _), (b, _)| a.cmp(b));
+
+    let last_idx = keyed_list.len() - 1;
+    let mut current_key = keyed_list[0].0.clone();
+    let mut current_ctn = 0;
+    let mut limit_message = String::new();
+    keyed_list
+        .drain(..)
+        .enumerate()
+        .filter_map(|(idx, (key, warning))| {
+            if key != current_key || idx == last_idx {
+                if current_ctn >= PER_LINT_LIMIT {
+                    writeln!(
+                        &mut limit_message,
+                        "* Hid {} messages for `{}`",
```

Comment:
I also wasn't sure what a good message would be. I'm open to alternative suggestions :D

Reply:
Could call it truncated

Reply:
Truncated, that's the perfect word :D